iox_query/src/provider.rs

```diff
@@ -72,6 +72,9 @@ pub enum Error {
 
     #[snafu(display("Internal error: Can not group chunks '{}'", source,))]
     InternalChunkGrouping { source: self::overlap::Error },
+
+    #[snafu(display("Too much data."))]
```

Comment:
Since this will be returned to the user, I feel like some more specifics are warranted:

```suggestion
    #[snafu(display("Query would scan {} bytes, more than configured maximum {} bytes. Try adjusting your compactor settings or increasing the per query memory limit.", maximum_size_limit))]
```